tiledb/sm/query/reader.cc

```diff
@@ -1294,7 +1300,17 @@ Status Reader::copy_partitioned_var_cells(
       auto var_dest = buffer_var + var_offset;
 
       // Copy offset
-      std::memcpy(offset_dest, &var_offset, offset_size);
+      if (cfg_offsets_format == "bytes") {
+        std::memcpy(offset_dest, &var_offset, offset_size);
+      } else {
+        assert(cfg_offsets_format == "elements");
+        if (fill_value_size == 1) {
+          std::memcpy(offset_dest, &var_offset, offset_size);
+        } else {
+          uint64_t element_offset = var_offset / fill_value_size;
```

Comment:
I think this will work most of the time, but let's use the size of the value rather than the fill value size.

Add this line to L1249: `auto attr_datatype_size = datatype_size(array_schema_->type(*name));` and then use `attr_datatype_size` here instead of `fill_value_size`.